internal/tiltfile/docker_compose.go

```diff
@@ -293,6 +282,56 @@ func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, c
 	return services, nil
 }
 
+func getConfigAndServiceNames(ctx context.Context, dcc dockercompose.DockerComposeClient,
+	configPath string) (conf dcConfig, svcNames []string, err error) {
+	// calls to `docker-compose config` take a bit, and we need two,
+	// so do them in parallel to make things faster
+	configGot := make(chan error, 1)
+	servicesGot := make(chan error, 1)
+
+	go func() {
+		configOut, err := dcc.Config(ctx, configPath)
+		if err != nil {
+			configGot <- err
+		}
+
+		err = yaml.Unmarshal([]byte(configOut), &conf)
+		if err != nil {
+			configGot <- err
+		}
+		close(configGot)
+	}()
+
+	go func() {
+		var err error
+		svcNames, err = serviceNames(ctx, dcc, configPath)
+		if err != nil {
+			servicesGot <- err
+		}
+		close(servicesGot)
+	}()
+
+	select {
+	case err := <-configGot:
+		if err != nil {
+			return conf, svcNames, err
+		}
+	case <-time.After(time.Second * 2):
```

Comment:
it seems weird to have a timeout here when there wasn't a timeout before? also 2s is shorter than some "normal" times i've seen

Reply:
yeahhh i just got antsy about leaving a `<- chan` in the middle of the code? but you're right that the previous behavior wasn't any safer in that regard.